### DISCUSSION TITLE
Not using cloud vs Spectre/Nordigen

### DISCUSSION THREAD

**Discussion by mvdkleijn** [firefly-iii/firefly-iii#5455] [Firefly III]
First of all, thanks for Firefly III! This is a sincere question though, Firefly III seems like a really nice piece of software with attention to detail so I'm surprised about the following apparent inconsistency? (not sure that's the right word here)

One of the major stated goals of Firefly III is to prevent people from having to upload their data to the cloud.

How does that fit with the use of services like Spectre & Nordigen which will get access to your data?
Especially considering the apparent lack of local importers?

Also, is there a list of importers somewhere? That'd really help. I have ABN AMRO and their CSV format is horrible. Important information like counter account is stuck in the description field so I'd love to find any alternative importers that can truly handle that.

**Comment by JC5** [comment]
Hey!

Great question. Short answer: I don't know, but I don't consider it an inconsistency. Using Firefly III is easy:
  
- Run Firefly III entirely offline.
- Create all transactions by hand.
  
This is what I do. I don't use any importer tool myself, even though I build them.

---

If you want to import (historical) data you can do a few things.

- Get (shitty) CSV files from your bank and import those throught the data importer. This is entirely offline.
- Use the provided cloud services, that automatically clean up your transactions (though ABN AMRO still sucks) and import those. This is "online".

This is not an inconsistency per se, but just the state of things. There are more than 12 banks in the Netherlands alone, all with their own (proprietary) data format. Even those banks that use MT940 or ISO 20022 use it in wildly different ways. And [unfortunately](https://docs.firefly-iii.org/firefly-iii/faq/general/#will-firefly-iii-support-psd2), PSD2 does not help. Each bank has its own format and way of doing things.

Is there a list of importers? [Yes](https://docs.firefly-iii.org/firefly-iii/importing-data/introduction/), but it doesn't yet list "ABN AMRO importer" or "Rabobank importer", unless you build one :wink:

Remember, you don't need to import anything. The use of Spectre and/or Nordigen is entirely up to you, and I'm happy to oblige (and I like the challenge). If you connect Spectre or Nordigen to your ABN account, that's not my doing.

---

If you're not happy about the "lack of local importers", I agree. But I can only suggest you build it yourself.

I can't build specific importers for each bank. I don't have the time or the resources to do so. But, if you write a Python script (or something similar) to clean their CSV files, I'm happy to list it together with its [import configuration](https://github.com/firefly-iii/import-configurations/tree/main/nl).

**Reply by mvdkleijn** [reply]
Hmmm, creating all transactions by hand like you do is not something I would have time for I think. Next to all the transactions of my bank, I would hope to eventually include Paypal transactions and crypto transactions. I'll be working on a CSV tool for ABN AMRO I guess. :smile:

**Comment by MacPaille** [comment]
I don't think using Nordigen/Spectre will really help. I also have this with Deutsche Bank in Belgium where they just don't list the other party account except in the description... Only way is to use rules to sort this out correctly when you import the transactions... Either using a CSV or one of these services.

**Comment by marceltc** [comment]
I think while having the online importers is a great feature improvement, I wonder if anyone else has concerns about privacy. The Spectre privacy policy says they include financial transactions as part of the personal data that they use for research, data analytics, etc. Does that mean by using these importers with Firefly III we're giving up our financial transaction info?

**Reply by JC5** [reply]
Yes. There is no exception for Firefly III users, and it's optional for a reason.

**Reply by mvdkleijn** [reply]
Generic comment: It's nice that the Spectre and Nordigen options exist, but I'd personally say that anyone concerned with their privacy should not be using them. If someone were to use Firefly III for privacy reasons (a common self host reason), then using these services would kind of defeat the purpose.

While I understand you add transactions by hand @JC5, and that's your choice & perfectly fine of course, it is not an option for a lot of folks. Me included. It just takes too much time and could introduce errors.

I've been having a go at writing a script to cleanup the ABNAMRO CSV file so that could get imported using the FIDI tool, but I'm getting inconsistent results. Thinking that I did something wrong, I checked both the cleaned up CSV file as well as the original TAB (as they call it) from ABNAMRO. Honestly, they seem fine.

I'll be double-checking my work again later this week but what baffles me is: in one import it appears to work fine and in another import (of the same data) after I flush the accounts and transactions, I suddenly get double transactions or other errors. I ran the importer several times like that to test both myself and the system but I cannot explain the differences.

Anyway, I'll look at those again later this week. If I still cannot get sane results, I'll try using the API directly instead of the FIDI tool and see if that works.

**Reply by JC5** [reply]
If you have something that you can reproduce, please send it to me so I can fix it.
